src/supplemental/http/http_public.c

```diff
@@ -779,6 +779,18 @@ nng_http_server_get_tls(nng_http_server *srv, struct nng_tls_config **cfgp)
 #endif
 }
 
+int
+nng_http_server_get_port(nng_http_server *srv, int *portp)
```

Comment:
The port numbers should be uint16_t.

It might be better to have a way to get the local address.  We have NNG_OPT_LOCADDR for other places in the code that use dynamically allocated ports like this.

Reply:
You mean something similar to nng_stream_listener_get_int?

Reply:
More like nng_stream_listener_get_addr().  This would give you the entire address (IP address and port) for the server.

The reason the server gets the port under the hood is to support websocket consumers that want to know the details.

Actually, Right now we can do this for websocket via the private nni_http_server_getx() API --
because websocket listeners (which are just HTTP servers) use this to support NNG_OPT_LOCADDR.

I'm about to commit test code which demonstrates this quite nicely.

Reply:
Ok, I'll wait for that and update the PR :)